src/components/select-field/BaseSelectField.js

```diff
@@ -221,6 +250,11 @@ class BaseSelectField extends React.Component<Props, State> {
                 break;
             case 'Enter':
             case ' ':
+                // When the search input is active, we want to allow space key presses in the search string
+                if (key === ' ' && shouldShowSearchInput) {
```

Comment:
When the user wants to use the SearchForm, they will likely enter in a space keypress. Thus, we should allow space keypresses to go through rather than either "selecting an option" or "closing the dropdown". This is why we now have an early `break` statement.